agent/client/client.go

```diff
@@ -0,0 +1,96 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/kubeshop/tracetest/agent/proto"
+	"google.golang.org/grpc"
+)
+
+type Config struct {
+	APIKey    string
+	AgentName string
+}
+
+type SessionConfig struct {
+	BatchTimeout time.Duration
+}
+
+type Client struct {
+	conn          *grpc.ClientConn
+	config        Config
+	sessionConfig *SessionConfig
+	done          chan bool
+
+	triggerListener func(*proto.TriggerRequest) error
```

Comment:
Decided to use a callback instead of a channel because this way we don't freeze the agent when there's more than one request at the same time.